spec/views/transactions/index.html.erb_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe 'transactions/index', type: :view do
+  before(:each) do
+    assign(:budget_transactions, [])
+    assign(:category, create(:category, name: 'Category Name'))
+  end
+
+  context 'when there are no transactions' do
+    it 'displays a message if there are no transactions' do
+      render
+      expect(rendered).to have_content('Transactions for Category Name')
+      expect(rendered).to have_content('Total Amount:')
+    end
+  end
+end

```

Comment:
- You have done well so far, but according to the project requirements, you are to write comprehensive integration tests to ensure your app functions properly. Please fix this 🙏